src/components/dev-hub/feedback/feedback-container.js

```diff
@@ -11,21 +18,111 @@ const feedbackItems = graphql`
     }
 `;
 
-const FeedbackContainer = () => {
-    const data = useStaticQuery(feedbackItems);
+const StyledButtonContainer = styled('div')`
+    display: flex;
+    justify-content: center;
+    margin-bottom: 4px;
+`;
+
+const StyledForm = styled('form')`
+    padding: 0 ${size.medium};
+
+    button,
+    label {
+        margin: ${size.default} 0;
+    }
+
+    ${InputContainer} {
+        margin: ${size.default} 0;
+    }
+`;
+
+const StyledDescription = styled(P)`
+    display: inline-block;
+    font-family: Akzidenz;
+    font-size: ${fontSize.small};
+    line-height: ${lineHeight.small};
+`;
+
+const StyledHeader = styled(H5)`
+    margin-bottom: 0;
+`;
+
+const getStarRatingFlow = (data, star, step) => {
     const {
         OneStarRatingFlow,
         TwoStarRatingFlow,
         ThreeStarRatingFlow,
         FourStarRatingFlow,
         FiveStarRatingFlow,
     } = dlv(data, 'strapiFeedbackRatingFlow', {});
-    // Uncomment below to see shape of data, other forms
-    // console.log(data);
-    // TODO: Implement feedback UI
-    return (
-        <CMSForm form={OneStarRatingFlow ? OneStarRatingFlow.forms[1] : []} />
-    );
+
+    switch (star) {
+        case 'one':
+            return OneStarRatingFlow?.forms[step];
+        case 'two':
+            return TwoStarRatingFlow?.forms[step];
+        case 'three':
+            return ThreeStarRatingFlow?.forms[step];
+        case 'four':
+            return FourStarRatingFlow?.forms[step];
+        case 'five':
+            return FiveStarRatingFlow?.forms[step];
+        default:
+            return [];
+    }
+};
+
+const FeedbackContainer = ({ onSubmit, starRatingFlow }) => {
+    const [step, setStep] = useState(0);
+    const data = useStaticQuery(feedbackItems);
+
+    const ratingFlow = getStarRatingFlow(data, starRatingFlow, step);
+
+    const title = ratingFlow?.title;
+    const description = ratingFlow?.description;
+    const button = ratingFlow?.cta;
+
+    const isActiveModal = step < 2;
+
+    return isActiveModal ? (
+        <Modal
+            isOpenToStart={true}
+            verticallyCenter
+            contentStyle={{
+                'max-width': '400px',
+                'min-width': '400px',
+            }}
+        >
+            <StyledForm
+                onSubmit={e => {
+                    e.preventDefault();
+                    onSubmit('Some value');
+                    setStep(prevState => prevState + 1);
```

Comment:
```suggestion
                    setStep(step + 1);
```

Would this be cleaner?

Reply:
Yeah, looks cleaner and for this simple state will be better, but I usually use variant with function because 'prevState' is our state in 100%.

Reply:
Okay sounds good to me